File: CodingInterviewsPackSolutionsCPP/392. Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int ptr1 = 0;
        int ptr2 = 0;
        int count = 0;
        while(ptr1<s.size() && ptr2<t.size()){
            if(s[ptr1]==t[ptr2]){
                count++;
                ptr1++;
                ptr2++;
            }else
                ptr2++;
        }
        return count==s.size();
    }
};

File: CodingInterviewsPythonSolutions/392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ptr1 = 0
        ptr2 = 0
        count = 0
        
        while ptr1 < len(s) and ptr2 < len(t):
            if s[ptr1] == t[ptr2]:
                count += 1
                ptr1 += 1
                ptr2 += 1
            else:
                ptr2 += 1
        
        return count == len(s)
